python_modules/dagster/dagster/_core/execution/asset_backfill.py

```diff
@@ -693,14 +705,22 @@ def _submit_runs_and_update_backfill_in_chunks(
         asset_graph=asset_graph,
         logger=logger,
         debug_crash_flags={},
+        backfill_id=backfill_id,
+        catch_code_location_load_errors=True,
     ):
-        if run_requests_chunk is None:
+        if submit_run_request_chunk_result is None:
             # allow the daemon to heartbeat
             yield None
             continue
 
+        code_unreachable_error_raised = (
+            submit_run_request_chunk_result.code_unreachable_error_raised
```

Comment:
this isn't neccesarily related to this PR specifically, but I found the logic around this for loop pretty hard to wrap my head around - in particular the fact that it will only ever return None until it returns something, then we can guarantee on that something being the last iteration (believe I have that right?) Without that understanding it seemed like things were going to just keep on going (within the same iteration) even after one of the runs failed to submit, but that is not the case - once the submissions either finish or raise this particular error, the iteration is over

One way to make that more clear would be to move most of the logic here outside of the for loop - to make it clear that once submit_run_request_chunk_result is not None, the for loop is guaranteed to be finished

Reply:
> it will only ever return None until it returns something, then we can guarantee on that something being the last iteration (believe I have that right?)

This is a confusing loop... it returning something doesn't mean it is actually the last iteration though.

The function will return a result for each run request chunk, yielding many `None`s between each run request chunk to all the daemon to heartbeat.

> once the submissions either finish or raise this particular error, the iteration is over

This part is true though, enabled by the break statements

Let me know if there are other ways you can imagine to make this more digestible and I can file a follow-up PR.